### PR TITLE
Display GAPs events in the Activity and Map views // VV-118

### PR DESCRIPTION
**Pull request by fertandil87**
WIP

image
image

### CODE REVIEW

File: apps/vessel-history/features/dataviews/dataviews.config.ts [hunk 1]
```diff
@@ -54,11 +54,11 @@ export const OFFLINE_LAYERS: GeneratorConfig[] = [
 export const DEFAULT_BASEMAP_DATAVIEW_ID = WORKSPACE_ENV === 'development' ? 90 : 173
 // export const DEFAULT_VESSEL_DATAVIEW_ID = WORKSPACE_ENV === 'development' ? 291 : 268 // Vessel with port visits c3 and c4
 export const DEFAULT_VESSEL_DATAVIEWS = {
-  'port-inspector': WORKSPACE_ENV === 'development' ? 176 : 268, // Vessel with port visits c2
+  'port-inspector': WORKSPACE_ENV === 'development' ? 291 : 268, // Vessel with port visits c2
   'insurance-underwriter': WORKSPACE_ENV === 'development' ? 291 : 304, // Vessel with port visits c3 and c4
 }
 export const DEFAULT_EEZ_DATAVIEW_ID = WORKSPACE_ENV === 'development' ? 94 : 177
-export const DEFAULT_MPA_DATAVIEW_ID = WORKSPACE_ENV === 'development' ? 98 : 176
+export const DEFAULT_MPA_DATAVIEW_ID = WORKSPACE_ENV === 'development' ? 98 : 291
```
Comment (rdgfuentes) [inline apps/vessel-history/features/dataviews/dataviews.config.ts:61]: Remember to revert this change, MPA layer should not change, right?

File: apps/vessel-history/features/dataviews/dataviews.config.ts [hunk 2]
```diff
@@ -54,11 +54,11 @@ export const OFFLINE_LAYERS: GeneratorConfig[] = [
 export const DEFAULT_BASEMAP_DATAVIEW_ID = WORKSPACE_ENV === 'development' ? 90 : 173
 // export const DEFAULT_VESSEL_DATAVIEW_ID = WORKSPACE_ENV === 'development' ? 291 : 268 // Vessel with port visits c3 and c4
 export const DEFAULT_VESSEL_DATAVIEWS = {
-  'port-inspector': WORKSPACE_ENV === 'development' ? 176 : 268, // Vessel with port visits c2
+  'port-inspector': WORKSPACE_ENV === 'development' ? 291 : 268, // Vessel with port visits c2
```
Comment (rdgfuentes) [inline apps/vessel-history/features/dataviews/dataviews.config.ts:57]: port inspectors should still use the dataviews with port visits c2 (176 and 268)

Comment (fertandil87) [reply apps/vessel-history/features/dataviews/dataviews.config.ts:57]: yes

File: apps/vessel-history/features/vessels/activity/vessels-activity.selectors.ts [hunk 3]
```diff
@@ -112,11 +112,44 @@ export const selectEventsForTracks = createSelector(
   }
 )
 
+// This selet will split the gaps in two events before add the rendering data
+export const selectEventsWithGapsSplit = createSelector([selectEventsForTracks], (events) => {
+  return events // I dont know if this change is temporaly
```
Comment (rdgfuentes) [inline apps/vessel-history/features/vessels/activity/vessels-activity.selectors.ts:117]: @fertandil87 if this selector only returns events it's not needed, please remove it.

File: apps/vessel-history/features/resources/resources.slice.ts [hunk 4]
```diff
@@ -38,13 +38,28 @@ export const selectResources = createSelector([originalSelectResource], (resourc
             start: event.end as number,
             id: `${event.id}-exit`,
           }))
-
+      /*
```
Comment (rdgfuentes) [inline apps/vessel-history/features/resources/resources.slice.ts:41]: @fertandil87 Remove this commented code 🧹

File: apps/vessel-history/features/resources/resources.slice.ts [hunk 5]
```diff
@@ -38,13 +38,28 @@ export const selectResources = createSelector([originalSelectResource], (resourc
...
+            id: `${event.id}-end`,
+          }))
+          */
       return [
         url,
         {
           ...resource,
           data: Array.isArray(resource.data)
-            ? (resource.data as any[])?.concat(portExitEvents)
+            ? (resource.data as any[])?.concat(portExitEvents)//.concat(gapsEnds)
```
Comment (rdgfuentes) [inline apps/vessel-history/features/resources/resources.slice.ts:62]: @fertandil87 Same here with `//.concat(gapsEnds)`

File: apps/vessel-history/features/vessels/activity/vessels-activity.selectors.ts [hunk 6]
```diff
@@ -173,6 +206,19 @@ export const selectEventsWithRenderingInfo = createSelector(
...
             descriptionGeneric = t('event.fishing')
             break
+          case EventTypes.Gap:
+            /*
+            Will split the gaps? 
+            description = event.gap.isEventStart ? t('event.gapStart', 'Gap start in {{regionName}}', {
+              regionName: regionDescription,
+            }) : t('event.gapEnd', 'Gap end in {{regionName}}', {
+              regionName: regionDescription,
+            }) */
```
Comment (rdgfuentes) [inline apps/vessel-history/features/vessels/activity/vessels-activity.selectors.ts:216]: @fertandil87 Could you please remove this commented code block? if we don't need it now it's better to clean this up so the code is more readable and we don't have to think if it is valid, if we have to maintain it, if can or can't be removed...
